engine-standalone-storage/src/engine_state.rs

```diff
@@ -49,8 +51,8 @@ impl<'db, 'input, 'output> EngineStateAccess<'db, 'input, 'output> {
         input: &'input [u8],
         bound_block_height: u64,
         bound_tx_position: u16,
-        transaction_diff: &'output RefCell<Diff>,
-        output: &'output Cell<Vec<u8>>,
+        transaction_diff: &'output Mutex<Diff>,
+        output: &'output OnceLock<Vec<u8>>,
```

Comment:
I don't know if we should use `OnceLock` here. I think it is probably true that `output` is only set once in each transaction, but I don't know if we can really guarantee it. I think we should keep the previous behaviour of only keeping the last `output`.